docs/ARCHITECTURE.md

```diff
@@ -0,0 +1,207 @@
+# Architecture
+
+:warning: **WORK IN PROGRESS**
+
+The Starter Kit architecture is designed to support scalable, modularised applications. Built around [Redux](http://redux.js.org/), it makes it simple to reason about your application's state, and as a result to write maintainable, error-free programs.
+
+The downside of the Starter Kit architecture is that it involves a number of novel concepts, which might take a while to grok for the uninitiated. This document aims to explain the what, why and how of building apps the Starter Kit way.
+
+## Pieces of the puzzle
+
+ * [Redux](http://redux.js.org/)
+ * [redux-loop](https://github.com/raisemarketplace/redux-loop)
+ * [ImmutableJS](https://facebook.github.io/immutable-js)
+
+The application state and state changes are managed by **Redux**, a library that implements a pure, side-effect-free variant of the Facebook Flux architecture. Redux and Flux prescribe a unidirectional dataflow through your application. To understand Redux, check out this [Cartoon guide by Lin Clark](https://code-cartoons.com/a-cartoon-guide-to-flux-6157355ab207#.4dpmozm9v) (it's great, not a joke!) and [Dan Abramov's Redux course on egghead.io](https://egghead.io/series/getting-started-with-redux).
+
+Redux helps us with synchronous updating of our state, but it doesn't provide an out-of-the-box solution for handling asynchronous actions. The Redux ecosystem has many possible solutions for this problem. In our application we use the vanilla redux-thunk middleware for simple asynchronous actions, and **redux-loop** to handle more complex asynchronisity.
+
+The state in Redux applications should never be mutated, but always cloned. To make this more natural for the programmer, and more-fault tolerant against accidental mutation, we use **ImmutableJS** data structures to hold our app's state.
+
+## Organising code
+
+Let's take a look of how we organise our application.
```

Comment:
we could add a directory structure here for a general overview (or link to our main folder in the code)